infrastructure/src/main/java/org/corfudb/infrastructure/configuration/ServerConfiguration.java

```diff
@@ -0,0 +1,475 @@
+        setProperty(HOST_ADDRESS, address);
+        return this;
+    }
+
+    public String getHostAddress() {
+        return getString(HOST_ADDRESS);
+    }
+
+    public ServerConfiguration setServerPort(int port) {
+        setProperty(SERVER_PORT, port);
+        return this;
+    }
+
+    public String getLocalServerEndpoint() {
+        return getHostAddress() + ":" + getServerPort();
+    }
+
+    public int getServerPort() {
+        return getInt(SERVER_PORT);
+    }
+
+    public ServerConfiguration setNetworkInterface(String networkInterface) {
+        setProperty(NETWORK_INTERFACE, networkInterface);
+        return this;
+    }
+
+    public String getNetworkInterface() {
+        return getString(NETWORK_INTERFACE);
+    }
+
+    public ServerConfiguration setHandshakeTimeout(int timeout) {
+        setProperty(HANDSHAKE_TIMEOUT, timeout);
+        return this;
+    }
+
+    public int getHandshakeTimeout() {
+        return getInt(HANDSHAKE_TIMEOUT, 10);
+    }
+
+    public ServerConfiguration setMetadataRetention(int numFiles) {
+        if (numFiles < 1) {
+            throw new IllegalArgumentException("Max number of metadata files to retain must be greater than 0.");
+        }
+        setProperty(METADATA_RETENTION, numFiles);
+        return this;
+    }
+
+    public int getMetadataRetention() {
+        return getInt(METADATA_RETENTION, 1000);
+    }
+
+    public ServerConfiguration setLogLevel(String level) {
+        setProperty(LOG_LEVEL, level);
+        return this;
+    }
+
+    public String getLogLevel() {
+        return getString(LOG_LEVEL, "INFO");
+    }
+
+    public ServerConfiguration setNumBaseServerThreads(int numThreads) {
+        setProperty(NUM_BASE_SERVER_THREADS, numThreads);
+        return this;
+    }
+
+    public int getNumBaseServerThreads() {
+        return getInt(NUM_BASE_SERVER_THREADS, 4);
+    }
+
+    public ServerConfiguration setMetricsProviderAddress(String address) {
+        setProperty(METRICS_PROVIDER_ADDRESS, address);
+        return this;
+    }
+
+    public Integer getMetricsProviderAddress() {
+        return getInteger(METRICS_PROVIDER_ADDRESS, null);
+    }
+
+    public ServerConfiguration setChannelImplementation(ChannelImplementation type) {
+        setProperty(CHANNEL_IMPLEMENTATION, type);
+        return this;
+    }
+
+    public ChannelImplementation getChannelImplementation() {
+        if (getProperty(CHANNEL_IMPLEMENTATION) == null) {
+            return ChannelImplementation.NIO;
+        } else {
+            return (ChannelImplementation) getProperty(CHANNEL_IMPLEMENTATION);
+        }
+    }
+
+    public boolean getBindToAllInterfaces() {
+        if (getNetworkInterface() != null || getHostAddress() != null) {
+            return false;
+        } else {
+            return true;
+        }
+    }
+
+    //todo(maithem) need to remove this, workaround library bug
+    Map<String, EventLoopGroup> testEventLoops = new HashMap<>();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://app.codacy.com/app/annymartinez.m/CorfuDB/pullRequest?prid=4651442)